src/app/settings/shelf-settings/shelf-settings.module.ts

```diff
@@ -0,0 +1,23 @@
+import { NgModule } from '@angular/core'
+import { SharedModule } from 'src/app/shared/shared.module'
+import { RouterModule, Routes } from '@angular/router'
+import { SettingsSharedModule } from '../settings-shared/settings-shared.module'
+import { ShelfActionPanelSettingsPageComponent } from './components/shelf-action-panel-settings-page/shelf-action-panel-settings-page.component'
+import { EditShelfActionPanelConfigurationComponent } from './components/edit-shelf-action-panel-configuration/edit-shelf-action-panel-configuration.component'
+import { EditShelfActionPanelConfigurationDialogComponent } from './components/edit-shelf-action-panel-confinguration-dialog/edit-shelf-action-panel-configuration-dialog.component'
+
+export enum ShelfSettingsPath {
+  ACTION_PANELS = 'action_panels',
```

Comment:
My understanding for routes is they must be `"-"` not `"_"`

Reply:
Good catch. I have updated it here and for `SettingsPath`.